File: retroauth-core/src/main/java/retrofit2/Retrofit2Platform.java
package retrofit2;

import java.util.concurrent.Executor;

import retrofit2.CallAdapter.Factory;

/**
 * This helper gets some os dependent information from retrofit.
 */
public final class Retrofit2Platform {

    private Retrofit2Platform() {
    }

    public static Factory defaultCallAdapterFactory(Executor executor) {
        if (executor == null) {
            executor = defaultCallbackExecutor();
        }
        return Platform.get().defaultCallAdapterFactory(executor);
    }

    public static Executor defaultCallbackExecutor() {
        return Platform.get().defaultCallbackExecutor();
    }
}

File: retroauth/src/main/java/retrofit2/Retrofit2Platform.kt
/*
 * Copyright (c) 2016 Andre Tietz
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package retrofit2

import retrofit2.CallAdapter.Factory
import java.util.concurrent.Executor

/**
 * This helper gets some os dependent information from retrofit.
 */
internal object Retrofit2Platform {
  @JvmStatic
  fun defaultCallAdapterFactories(executor: Executor?): List<Factory> {
    executor?.let {
      return Platform.get().defaultCallAdapterFactories(executor)
    }
    return Platform.get().defaultCallAdapterFactories(Platform.get().defaultCallbackExecutor())
  }
}
